SofascoreAcademy/SofascoreAcademy/Views/CustomTabBarController.swift

```diff
@@ -0,0 +1,134 @@
+        tab2View = createButton(logoString: "icon_basketball", title: "Basketball")
+        tab3View = createButton(logoString: "icon_american_football", title: "Am. Football")
+        
+        tabsContainerView.addSubview(tab1View)
+        tabsContainerView.addSubview(tab2View)
+        tabsContainerView.addSubview(tab3View)
+        tabsContainerView.addSubview(tabIndicator)
+    }
+    
+    override func styleViews() {
+        backgroundColor = colors.colorPrimaryDefault
+        tabIndicator.backgroundColor = .white
+        tabIndicator.layer.cornerRadius = 2
+    }
+    
+    override func setupConstraints() {
+        tabsContainerView.snp.makeConstraints() {
+            $0.edges.equalToSuperview()
+        }
+        
+        tabIndicator.snp.makeConstraints() {
+            $0.height.equalTo(4)
+            $0.width.equalTo(115)
+            $0.bottom.equalToSuperview()
+            $0.leading.equalTo((n-1)*131+8)
+        }
+        
+        let tabWidth = UIScreen.main.bounds.width/3
+        tab1View.snp.makeConstraints() {
+            $0.top.bottom.equalToSuperview()
+            $0.width.equalTo(tabWidth)
+            $0.leading.equalToSuperview()
+        }
+        
+        tab2View.snp.makeConstraints() {
+            $0.top.bottom.equalToSuperview()
+            $0.width.equalTo(tabWidth)
+            $0.leading.equalTo(tab1View.snp.trailing)
+        }
+        
+        tab3View.snp.makeConstraints() {
+            $0.top.bottom.equalToSuperview()
+            $0.width.equalTo(tabWidth)
+            $0.leading.equalTo(tab2View.snp.trailing)
+        }
+    }
+}
+
+extension CustomTabBarController{
+    
+    func createButton(logoString: String, title: String) -> UIButton {
+        let button = UIButton()
+        let buttonTitle = UILabel()
+        let buttonLogo = UIImageView()
+        
+        button.addSubview(buttonTitle)
+        button.addSubview(buttonLogo)
+        
+        buttonTitle.text = title
+        buttonTitle.textAlignment = .center
+        buttonTitle.textColor = .white
+        buttonTitle.font = fonts.RobotoRegular14
+        buttonLogo.image = UIImage(named: logoString)
+        
+        let tabWidth = UIScreen.main.bounds.width/3
+        
+        buttonLogo.snp.makeConstraints() {
+            $0.size.equalTo(16)
+            $0.top.equalToSuperview().offset(4)
+            $0.leading.equalToSuperview().offset(tabWidth/2-8)
+        }
+        
+        buttonTitle.snp.makeConstraints() {
+            $0.leading.trailing.equalToSuperview().inset(8)
+            $0.bottom.equalToSuperview().inset(8)
+            $0.top.equalToSuperview().offset(24)
+        }
+        
+        button.addTarget(self, action: #selector(buttonClicked), for: .touchUpInside)
+        
+        return button
+    }
+}
+
+extension CustomTabBarController {
+    
+    @objc func buttonClicked(_ sender: UIButton) {
+        var data: sport_slug?
```

Comment:
Kad napravis TabViewItem on bi ti ovdje trebao vratiti svoj sportSlug da izbjegnes ove if else-ove. 

Dakle, imat ces taj view novi koji ce ti zamjeniti ove tvoje buttone tab1View, tab2View... i taj view bi trebao na svoj tap javiti ovome tu CustomTabView-u hej, kliknuo si na mene a ja sam taj i taj sportSlug, to je njegov identifikator